kubernetes_tests/test_kubernetes_pod_operator.py

```diff
@@ -387,8 +369,53 @@ def test_pod_resources(self):
         }
         assert self.expected_pod == actual_pod
 
-    def test_pod_affinity(self):
-        affinity = {
+    @pytest.mark.parametrize(
+        "val",
+        [
+            param(
+                k8s.V1Affinity(
+                    node_affinity=k8s.V1NodeAffinity(
+                        required_during_scheduling_ignored_during_execution=k8s.V1NodeSelector(
+                            node_selector_terms=[
+                                k8s.V1NodeSelectorTerm(
+                                    match_expressions=[
+                                        k8s.V1NodeSelectorRequirement(
+                                            key="beta.kubernetes.io/os",
+                                            operator="In",
+                                            values=["linux"],
+                                        )
+                                    ]
+                                )
+                            ]
+                        )
+                    )
```

Comment:
OMG this is ugly (not your fault)

Reply:
yes, quite. makes you wonder whether it really makes sense to deprecate the old way

Reply:
Ugly it is. I agree. What are the benefits in deprecating the old way BTW :)

Reply:
well, this was before my time, but obvs if you're using objects it can be helpful e.g. with autocomplete and validation.  
but yeah it was actually kind of tricky to use. 

maybe we should only be deprecating the airflow objects, and still support json or k8s object.

but, note that this PR doesn't change anything related to deprecations.  it just gets rid of noise in the "backcompat" test, and moves legitimate "backcompat" tests inline with the main tests. which helps e.g. in this case becasue you can see side-by-side new vs "backcompat". in this particular test, there actually _wasn't_ any test of the "new" behavior previously (maybe because it's such a pain in the ass?) so i added that here.

anyway, if any of you "looky lous" has an inclination to approve that would be welcome